internal/monitor/monitor.go

```diff
@@ -135,18 +155,48 @@ func (pm *PowerMonitor) initZones() error {
 	return nil
 }
 
+// collectionLoop handles periodic data collection
+func (pm *PowerMonitor) collectionLoop() {
+	if err := pm.synchronizedPowerRefresh(); err != nil {
+		pm.logger.Error("Failed to collect initial power data", "error", err)
+	}
+
+	if pm.interval > 0 {
+		pm.scheduleNextCollection()
+	}
+}
+
+// scheduleNextCollection schedules the next data collection
+func (pm *PowerMonitor) scheduleNextCollection() {
+	timer := pm.clock.After(pm.interval)
+	go func() {
+		select {
+		case <-timer:
```

Comment:
we can get the current time from this channel, can avoid calling `Now()` in refresh data

Reply:
Now() should refer to run and not when the collection started, so caching/ reusing the current time isn't IMHO the right approach